pkg/deploy/lattice/service_manager.go

```diff
@@ -204,29 +204,28 @@ func (m *defaultServiceManager) getAllAssociations(ctx context.Context, svcSum *
 func (m *defaultServiceManager) updateAssociations(ctx context.Context, svc *Service, svcSum *SvcSummary) error {
 	assocs, err := m.getAllAssociations(ctx, svcSum)
 	if err != nil {
-		return err
+		return fmt.Errorf("in updateAssociations, getAllAssociations failed with %w", err)
 	}
 
 	toCreate, toDelete, err := associationsDiff(svc, assocs)
 	if err != nil {
-		return err
+		return fmt.Errorf("in updateAssociations, associationsDiff failed with %w", err)
 	}
 	for _, snName := range toCreate {
 		err := m.createAssociation(ctx, svcSum.Id, snName)
 		if err != nil {
-			return err
+			return fmt.Errorf("in updateAssociations, createAssociations failed with %w", err)
 		}
 	}
 
 	for _, assoc := range toDelete {
 		isManaged, err := m.cloud.IsArnManaged(ctx, *assoc.Arn)
 		if err != nil {
-			return err
-		}
-		if isManaged {
+			m.log.Errorf("in updateAssociations failed when attempting IsArnManaged check. Skipping delete association. service: %s, association: %s, %s", svc.LatticeServiceName(), assoc.Arn, err)
+		} else if isManaged {
 			err = m.deleteAssociation(ctx, assoc.Arn)
```

Comment:
should we mimic [this logic](https://github.com/aws/aws-application-networking-k8s/blob/144017002783c07c07ace72d14718a5fbadadac3/pkg/aws/services/vpclattice.go#L391-L393) to only ignore get tags AccessDenied and still return other err? for example:

```
	for _, assoc := range toDelete {
		isManaged, err := m.cloud.IsArnManaged(ctx, *assoc.Arn)
		if err != nil {
			aerr, ok := err.(awserr.Error)
			if ok && aerr.Code() != vpclattice.ErrCodeAccessDeniedException {
				// In a scenario that the service association is created by a foreign account,
				// the owner account's controller cannot read the tags of this ServiceNetworkServiceAssociation,
				// and AccessDeniedException is expected.
				continue
			} else {
				return err
			}
		}
		if isManaged {
			err = m.deleteAssociation(ctx, assoc.Arn)
			if err != nil {
				return err
			}
		}
	}
```

@solmonk  can you review this PR?

Reply:
yes, I think this way makes more sense in terms of consistency.